tgui/packages/tgui/interfaces/BarberingMenu.js

```diff
@@ -0,0 +1,119 @@
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Section, Image, Stack, ByondUi, Input, Icon } from '../components';
+import { Window } from '../layouts';
+
+export const BarberingMenu = (props, context) => {
+  const { data } = useBackend(context);
+  const { available_styles } = data;
+  const [searchText, setSearchText] = useLocalState(context, 'searchText', '');
+
+  const styles_keys = Object.keys(available_styles);
+
+  return (
+    <Window
+      width={700}
+      height={500}
+      title="Barber">
+
+      <Window.Content scrollable>
+        <HairOptions />
+        <hr />
+        <Stack>
+          <Stack.Item width="65%">
+            <Stack wrap="wrap" justify="space-around">
+              <Stack width="100%" inline justify="space-around">
+                <Icon name="magnifying-glass" />
+                <Stack.Item mb="10px" grow>
+                  <Input width="100%" onInput={(e, value) => setSearchText(value)} />
+                </Stack.Item>
+              </Stack>
+              <HairPreviewList search_text={searchText}
+                all_hair_names={styles_keys}
+                all_hair_styles={available_styles} />
+            </Stack>
+          </Stack.Item>
+          <Stack.Item position="fixed" width="32%">
```

Comment:
Not a fan of two items with percentages totaling 97%. Implies layout hacks to me a little. I get that this item is `position: fixed` so it doesn't affect the layout of the earlier one. Seems like these aren't `Stack.Item`s at this point, perhaps?

I believe the `PreviewWindow` could be a fixed width; would using a fixed width prevent distortion issues as you increase/decrease width?

Reply:
I'll have to admit, this is indeed somewhat layout hacks. If I try to use up the other 3%, the `PreviewWindow` will get over the scroll bar. Although, I also don't feel comportable using fixed widths, since it will actually worsen distortion by re-sizing the window, because it forces the `PreviewWindow` to get over the scroll bar once again if there's too little space on the window. This is, as far as I know, the best solution to this issue.